src/renderer/window/window.cpp

```
@@ -42,20 +64,51 @@ mr::Window::Window(const RenderContext &parent, Extent extent)
+    _should_update_swapchain = true;
+    return std::nullopt;
+  }
+
+  CommandUnit command_unit {_parent->vulkan_state()};
+  command_unit.begin();
+  _swapchain._images[image_index].switch_layout(command_unit, vk::ImageLayout::eColorAttachmentOptimal);
+  command_unit.end();
 
-  _swapchain._images[image_index].switch_layout(_parent->vulkan_state(),
-                                                        vk::ImageLayout::eColorAttachmentOptimal);
+  UniqueFenceGuard(_parent->vulkan_state().device(), command_unit.submit(_parent->vulkan_state()));
```

Cppcheck (2.19.0)

[style] The function 'add_value' is never used.

(unusedFunction)